File: my_simple_robot_nodes/scripts/simple_number_publisher_with_param.py
#!/usr/bin/env python3
"""
    ROS publisher node publishing an incrementing number 
	Reads the publish rate from ROS Parameter server
    
    Author: Jari Honkanen
"""

import rospy
from std_msgs.msg import Int64

if __name__ == '__main__':

	rospy.init_node("simple_number_publisher")
	pub = rospy.Publisher("/number", Int64, queue_size=10)

	rospy.loginfo("[INFO] simple_number_publisher started")

	try:
		publish_freq = rospy.get_param("/number_publish_frequency")   # Rate in Hz
	except KeyError as e:
		rospy.logwarn("[ERROR] getting number_publish_frequency failed")
		publish_freq = 1

	rate = rospy.Rate(publish_freq) 

	rospy.set_param("/number_publisher_message", "Hi, there!")

	number = 1

	while not rospy.is_shutdown():

		msg = Int64()
		msg.data = number 
		pub.publish(msg)
		rospy.loginfo("[INFO] Published number " + str(number))
		number += 1
		rate.sleep()

File: my_simple_robot_nodes/src/simple_number_publisher_with_param.cpp
/*-------------------------------------------------------------------*\
  NAME
    simple_number_publisher.cpp

  DESCRIPTION
    ROS publisher node publishing an incrementing number 

  AUTHOR
    Jari Honkanen
\*-------------------------------------------------------------------*/

#include <ros/ros.h>
#include <std_msgs/Int64.h>

int main (int argc, char **argv)
{
	ros::init(argc, argv, "simple_number_publisher", ros::init_options::AnonymousName);
	ros::NodeHandle nh;

	ros::Publisher pub = nh.advertise<std_msgs::Int64>("/number", 10);

	double publish_freq;
	try {
		nh.getParam("/number_publish_frequency", publish_freq);
	}
	catch (const std::runtime_error &ex)
	{
		ROS_INFO("[ERROR] Reading of /number_publish_frequency failed.");
		publish_freq = 1;
	}
	ros::Rate rate(publish_freq); 

	nh.setParam("/number_publish_message", "Hi, there!");   

	ROS_INFO("[INFO] simple_number_publisher started");

	int number = 1;

	while (ros::ok()) {
		std_msgs::Int64 msg;
		msg.data = number;
		number++;
		pub.publish(msg);
		ROS_INFO("[INFO] Published number %d", number);
		rate.sleep();
	}
}
